dvc/api.py

```diff
@@ -134,34 +177,16 @@ def _get_object_from_summon_file(name, path):
 
         return objects[0]
 
-    except FileMissingError:
-        raise SummonError("Summon file not found")
+    except FileNotFoundError as exc:
+        raise SummonError("Summon file not found") from exc
     except ruamel.yaml.YAMLError as exc:
         raise SummonError("Failed to parse summon file") from exc
     except Invalid as exc:
-        raise SummonError(str(exc))
-
-
-def _pull_dependencies(repo, deps):
-    if not deps:
-        return
-
-    outs = [repo.find_out_by_relpath(dep) for dep in deps]
-
-    with repo.state:
-        for out in outs:
-            repo.cloud.pull(out.get_used_cache())
-            out.checkout()
+        raise SummonError(str(exc)) from exc
```

Comment:
I guess users of `summon` are not going to set up the DVC logger, right?
These messages would duplicate the output (`exc - exc`).
@efiop introduced an extra keyword to prevent that: `extra={"tb_only": True}`, if it is the case, maybe you could use it.

Reply:
The cause will have the same message as `SummonError` but different traceback, I don't think this is an issue. I can't use `extra={"tb_only": True}` because I am not using `logger.exception()` but simply raise an exception.

Reply:
@Suor I agree with @mroutis , it will look ugly that way. It is a formatting issue, right? Why not `SummonError("invalid formatting") from exc` or something like that?

Reply:
@efiop because a user won't see the issue until he/she scrolls up past traceback. "invalid formatting" is useless, you need to see, which key is absent/extraneous/has wrong type.

Reply:
Missed that it is not meant to go through our usual logger logic. I'm fine with it 👍